test/integration/targets/ec2_vpc_net/tasks/main.yml

```diff
@@ -33,194 +46,340 @@
+          - 'vpc_info.vpcs[0].cidr_block_association_set[0].cidr_block_state.state == "associated"'
+          - '"classic_link_dns_supported" in vpc_info.vpcs[0]'
+          - '"classic_link_enabled" in vpc_info.vpcs[0]'
+          - 'vpc_info.vpcs[0].dhcp_options_id == result.vpc.dhcp_options_id'
+          - '( vpc_info.vpcs[0].enable_dns_hostnames | bool ) == True'
+          - '( vpc_info.vpcs[0].enable_dns_support | bool ) == True'
+          - 'vpc_info.vpcs[0].id == result.vpc.id'
+          - '"instance_tenancy" in vpc_info.vpcs[0]'
+          - 'vpc_info.vpcs[0].ipv6_cidr_block_association_set | length == 1'
+          - 'vpc_info.vpcs[0].ipv6_cidr_block_association_set[0].association_id == result.vpc.ipv6_cidr_block_association_set[0].association_id'
+          - 'vpc_info.vpcs[0].ipv6_cidr_block_association_set[0].ipv6_cidr_block == result.vpc.ipv6_cidr_block_association_set[0].ipv6_cidr_block'
+          - 'vpc_info.vpcs[0].ipv6_cidr_block_association_set[0].ipv6_cidr_block_state.state == "associated"'
+          - '"is_default" in vpc_info.vpcs[0]'
+          - 'vpc_info.vpcs[0].owner_id == caller_facts.account'
+          - '"state" in vpc_info.vpcs[0]'
+          - 'vpc_info.vpcs[0].vpc_id == result.vpc.id'
+
+    # ============================================================
+
+    # This will only fail if there are already *2* vpcs otherwise ec2_vpc_net
+    # assumes you want to update your existing VPC...
     - name: attempt to create another VPC with the same CIDR and name without multi_ok
       ec2_vpc_net:
-        cidr_block: 20.0.0.0/24
-        ipv6_cidr: True
-        name: "{{ resource_prefix }}"
         state: present
+        cidr_block: "{{ vpc_cidr }}"
+        name: "{{ resource_prefix }}"
+        ipv6_cidr: True
+        tenancy: dedicated
         multi_ok: no
-        <<: *aws_connection_info
-      register: result
+      register: new_result
       ignore_errors: yes
+    - ec2_vpc_net_info:
+        filters:
+          "tag:Name": "{{ resource_prefix }}"
+      register: vpc_info
 
     - name: assert failure
       assert:
         that:
-          - 'result is failed'
-          - '"If you would like to create the VPC anyway please pass True to the multi_ok param" in result.msg'
+          - 'new_result is failed'
+          - '"If you would like to create the VPC anyway please pass True to the multi_ok param" in new_result.msg'
+          - 'vpc_info.vpcs | length == 2'
 
     # ============================================================
 
     # FIXME: right now if there are multiple matching VPCs they cannot be removed,
     # as there is no vpc_id option for idempotence. A workaround is to retag the VPC.
-    - name: remove Name tag on vpc_1
+    - name: remove Name tag on new VPC
       ec2_tag:
-        resource: "{{ vpc_1 }}"
         state: absent
+        resource: "{{ result.vpc.id }}"
```

Comment:
It was a bit challenging at this point to be certain which `result` was being referenced since that name gets reused so often and so many tests are being added in between where that was last registered and now being used.  I can see this being problematic for future test additions.  Could you please make the registered output name at ln261 more unique, maybe like `vpc_2_result`, and then update the various uses?

Reply:
I think this is the only place there's a distance between registration and use.

I've set a new fact "vpc_2" which contains the ID of this second VPC and use this rather than result.vpc.id